### PR TITLE
added download queue functionality

### PR DESCRIPTION
**Pull request by zlalvani**
The download queue API call should now work. There is also a running list of files on the "network" and files being downloaded.

### CODE REVIEW

**Comment by zlalvani** [comment]
pls review :(

**Comment by zlalvani** [comment]
pls

File: bin/modules/renter.dart [hunk 1]
```diff
@@ -2,15 +2,48 @@ library renter;
...
+
+class DownloadingFile extends UploadedFile{
+  
+  DownloadingFile(String nickname, int filesize, this.destination):
+    super(nickname, filesize);
+  
+  DownloadingFile.fromUploadedFile(UploadedFile file, this.destination):
+    super(file.nickname, file.filesize);
+
+  bool complete = false;
```
Comment (seveibar) [inline bin/modules/renter.dart:34]: varialbles should go at top- standard code convention

File: bin/modules/renter.dart [hunk 2]
```diff
@@ -46,6 +111,17 @@ class RegularRenter implements Renter{
   }
   
   shelf.Response DownloadQueue(shelf.Request req){
+    
+    List<Map> JSONmap = [];
```
Comment (seveibar) [inline bin/modules/renter.dart:115]: name doesn't describe what's in it

**Pull request merged**